dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java

```diff
@@ -103,6 +104,7 @@ public synchronized void registerWithMessage(EntityMessage invokeMessage, int co
     List<Retiree> toRetire = new ArrayList<>();
 
     LogicalSequence completedRequest = this.currentlyRunning.remove(completedMessage);
+    this.mostRecentRegisteredToKey.remove(completedRequest.concurrencyKey, completedRequest);
```

Comment:
Minor cleanup.  I don't want things hanging around in the map.  For example, if an entity decided to use a random int for the concurrency key, this would prevent a leak